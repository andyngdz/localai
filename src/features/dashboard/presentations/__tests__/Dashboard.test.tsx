import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Dashboard } from "../Dashboard";

describe("Dashboard", () => {
  it("renders dashboard component", () => {
    render(<Dashboard />);

    expect(screen.getByText("Dashboard")).toBeInTheDocument();
  });

  it("renders as a div element", () => {
    const { container } = render(<Dashboard />);

    const dashboardDiv = container.firstChild;
    expect(dashboardDiv).toBeInTheDocument();
    expect(dashboardDiv).toHaveTextContent("Dashboard");
  });

  it("has correct component structure", () => {
    const { container } = render(<Dashboard />);

    // Should render a single div element
    expect(container.children).toHaveLength(1);
    expect(container.firstChild?.nodeName).toBe("DIV");
  });

  it("matches snapshot", () => {
    const { container } = render(<Dashboard />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
