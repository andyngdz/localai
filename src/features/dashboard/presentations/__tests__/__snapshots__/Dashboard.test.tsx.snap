// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dashboard > matches snapshot 1`] = `
<div>
  Dashboard
</div>
`;
